storage/client.go

```diff
@@ -278,31 +278,33 @@ type newRangeReaderParams struct {
 
 type composeObjectRequest struct {
 	dstBucket     string
-	dstObject     composeDstObject
-	srcs          []composeSrcObject
+	dstObject     destinationObject
+	srcs          []sourceObject
 	predefinedACL string
 	encryptionKey []byte
 	sendCRC32C    bool
 }
 
-type composeSrcObject struct {
-	name  string
-	gen   int64
-	conds *Conditions
+type sourceObject struct {
+	name          string
+	bucket        string
+	gen           int64
+	conds         *Conditions
+	encryptionKey []byte
 }
 
-type composeDstObject struct {
-	name  string
-	conds *Conditions
-	attrs *ObjectAttrs // attrs to set on the destination object.
+type destinationObject struct {
+	name          string
+	bucket        string
+	conds         *Conditions
+	attrs         *ObjectAttrs // attrs to set on the destination object.
+	encryptionKey []byte
+	keyName       string
 }
 
 type rewriteObjectRequest struct {
-	srcBucket     string
-	srcObject     string
-	dstBucket     string
-	dstObject     string
-	dstKeyName    string
+	srcObject     sourceObject
+	dstObject     destinationObject
 	attrs         *ObjectAttrs
```

Comment:
I believe now you can get of `attrs` and `conds` in favor of the relevant `destinationObject` fields, and `gen` in favor of `sourceObject.gen`.

Reply:
Ah good call, I will do that.